src/graphics.c

```diff
@@ -616,10 +632,25 @@ void graphics_fill_screen( display_context_t disp, uint32_t c )
 }
 
 /**
- * @brief Draw a character to the screen using the built-in font
+ * @brief Set the current font. Should be set before using any of the draw function.
  *
- * @todo Need a set font mechanism.
- * @todo Need to look up the width and height from the font instead of assuming it is 8x8.
+ * @param[in] font
+ *        Sprite font to be used. Should be using a standard font sprite (containing every ASCII
+ * character) and with a transparent background, and be loaded on mksprite with width and height
+ * of 16.
+ * @param[in] tab_width
+ *        Width of each \\t used. Is multiplied by the width of each character.
+ */
+void graphics_set_font( sprite_t *font, int tab_width )
+{
+    sprite_font.sprite = font;
+    sprite_font.tab_width = tab_width;
```

Comment:
I know that tab_width is currently hardcoded but I think tab_width is more of a concern for text layouting / drawing routines and not specifically bound to a font. The fact that I want to use tabulations of 8 or 4 has more to do with how to want align the text and less with the used font (and the current implementation is also broken as the tab is just a substitute for a sequence of spaces rather than aligning to a certain column, but I digress).

If you don't have a pressing need on this, I would suggest to leave it out from the public API for now. You can keep it in the structure if you want with value 5, and then we can devise a better way to configure it and use it.

Reply:
I set that mostly for personal preference of a "normal" `tab_width` of 4 and not 5.

Do you have in mind how would we change this kind of setting in the future?

Reply:
Honestly I don't know. I think a more complex layouting API is due, as users would surely need to have their texts centered for instance, auto-split in multiple lines and so on. I think tabulation size is one possible options of this text layout function / engine. Also layouting is probably the same between graphics.c and rdp.c which is why it should probably be moved to a common module like a font.c.

An idea would be that this layout engine would render into a different buffer (another sprite) so that it can then be shown on screen as a simple textured quad. In this case, we would need some analysis to see whether its better to have a CPU-only renderer so that the quad can be I4/I8 for instance (after we support I4/I8 fonts) or it's better to try to use RDP and have this quad as RGB16. Rendering with RDP to a CI8 sprite is also an option which is worth investigating probably.

This is just a braindump of course, nothing concrete. Just to answer the question of why I think tabulation is more related to layout and less to the font.